core/src/main/resources/hudson/tasks/Shell/config.groovy

```diff
@@ -27,3 +27,11 @@ f=namespace(lib.FormTagLib)
 f.entry(title:_("Command"),description:_("description",rootURL)) {
     f.textarea(name: "command", value: instance?.command, class: "fixed-width", 'codemirror-mode': 'shell', 'codemirror-config': "mode: 'text/x-sh'")
 }
+
+f.advanced() {
+
+    f.entry(title:_("Return code to set build unstable"), description:_("If set, the script return code that will be interpreted as an unstable build result.")) {
+        f.number(name: "unstableReturn", value: instance?.unstableReturn, min:1, max:255, step:1)
```

Comment:
Where's the 'positive number' part present in `BatchFile`'s form? Not needed? Then why is it present there?